mesonbuild/modules/unstable_icestorm.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2017 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .. import mesonlib, compilers, mlog
+
+from . import ExtensionModule
+
+class IceStormModule(ExtensionModule):
+
+    def __init__(self):
+        super().__init__()
+        self.snippets.add('project')
+        self.yosys_bin = None
+
+    def detect_binaries(self, interpreter):
+        self.yosys_bin = interpreter.func_find_program(None, ['yosys'], {})
+        self.arachne_bin = interpreter.func_find_program(None, ['arachne-pnr'], {})
+        self.icepack_bin = interpreter.func_find_program(None, ['icepack'], {})
+        self.iceprog_bin = interpreter.func_find_program(None, ['iceprog'], {})
+        self.icetime_bin = interpreter.func_find_program(None, ['icetime'], {})
+
+    def project(self, interpreter, state, args, kwargs):
+        if not self.yosys_bin:
+            self.detect_binaries(interpreter)
+        result = []
```

Comment:
__[flake8]__
* [F841] local variable 'result' is assigned to but never used

<!-- Comment by [SideCI](https://sideci.com) -->